test_scripts/Capabilities/PersistingHMICapabilities/common.lua

```diff
@@ -0,0 +1,909 @@
+  local mobileConnection = actions.mobile.getConnection()
+  config.ExitOnCrash = false
+  local timeout = 5000
+  local function removeSessions()
+    for i = 1, actions.mobile.getAppsCount() do
+      actions.mobile.deleteSession(i)
+    end
+  end
+  local event = events.Event()
+  event.matches = function(event1, event2) return event1 == event2 end
+  mobileConnection:ExpectEvent(event, "SDL shutdown")
+  :Do(function()
+    removeSessions()
+    StopSDL()
+    config.ExitOnCrash = true
+  end)
+  hmiConnection:SendNotification("BasicCommunication.OnExitAllApplications", { reason = "SUSPEND" })
+  hmiConnection:ExpectNotification("BasicCommunication.OnSDLPersistenceComplete")
+  :Do(function()
+    hmiConnection:SendNotification("BasicCommunication.OnExitAllApplications",{ reason = "IGNITION_OFF" })
+    for i = 1, actions.mobile.getAppsCount() do
+      actions.mobile.getSession(i):ExpectNotification("OnAppInterfaceUnregistered", { reason = "IGNITION_OFF" })
+    end
+  end)
+  hmiConnection:ExpectNotification("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = false })
+  :Times(actions.mobile.getAppsCount())
+  local isSDLShutDownSuccessfully = false
+  hmiConnection:ExpectNotification("BasicCommunication.OnSDLClose")
+  :Do(function()
+    utils.cprint(35, "SDL was shutdown successfully")
+    isSDLShutDownSuccessfully = true
+    mobileConnection:RaiseEvent(event, event)
+  end)
+  :Timeout(timeout)
+  local function forceStopSDL()
+    if isSDLShutDownSuccessfully == false then
+      utils.cprint(35, "SDL was shutdown forcibly")
+      mobileConnection:RaiseEvent(event, event)
+    end
+  end
+  actions.run.runAfter(forceStopSDL, timeout + 500)
+end
+
+function m.start(pHMIParams)
+  local hmiParams = pHMIParams or hmi_values.getDefaultHMITable()
+  if hmiParams and hmiParams.UI and hmiParams.UI.GetCapabilities
+      and hmiParams.UI.GetCapabilities.params.displayCapabilities then
+    excludeAbsentInMobApiTextFields(hmiParams.UI.GetCapabilities.params.displayCapabilities.textFields)
+  end
+  return actions.start(hmiParams)
+end
+
+local function incompleteStart(isConnectMobile)
+  local event = actions.run.createEvent()
+  actions.init.SDL()
+  :Do(function()
+      local initHmiExp = actions.init.HMI()
+      if isConnectMobile then
+        initHmiExp:Do(function()
+            actions.init.connectMobile()
+            :Do(function()
+                actions.init.allowSDL()
+                :Do(function()
+                    actions.hmi.getConnection():RaiseEvent(event, "Start event")
+                  end)
+              end)
+          end)
+      else
+        initHmiExp:Do(function()
+            actions.hmi.getConnection():RaiseEvent(event, "Start event")
+        end)
+      end
+    end)
+  return actions.hmi.getConnection():ExpectEvent(event, "Start event")
+end
+
+function m.startWoHMIonReady()
+  return incompleteStart(true)
+end
+
+function m.startWoHMIonReadyAndMobile()
```

Comment:
```suggestion
function m.startWoHMIonReadyOrMobile()
```
this makes it more clear mobile will not be connected.

Reply:
@iCollin Name of function `startWoHMIonReadyAndMobile()` means **start** **W**ith**o**ut **HMI** **onReady** **And** without default **Mobile** connection.
Do you agree to leave existing function name?

Reply:
When I read it I thought it meant without OnReady but with Mobile, I think when you list multiple things negatively in english the word `or` is preferred over `and` but let me think about this one more, not a big deal either way :)

Reply:
@iCollin I understood. Thank you for explanation. 
To avoid confusion of us both I have rename the function to `startWoBothHMIonReadyAndMobile()` (see 2c89a37 )